commands.js

```diff
@@ -551,39 +551,28 @@ var commands = exports.commands = {
 		var targetId = toId(target) || user.userid;
 		var targetUser = Users.getExact(targetId);
 		var targetUsername = (targetUser ? targetUser.name : target);
+		var canPrivate = targetId === user.userid || user.can('makeroom');
 
 		var buffer = [];
-		var innerBuffer = [];
+		var publicAuth = [];
+		var privateAuth = [];
 		var group = Users.usergroups[targetId];
-		if (group) {
-			buffer.push('Global auth: ' + group.charAt(0));
-		}
-		for (var i = 0; i < Rooms.global.chatRooms.length; i++) {
+		if (group) buffer.push('Global auth: ' + group.charAt(0));
+
+		for (var i = 0, len = Rooms.global.chatRooms.length; i < len; i++) {
```

Comment:
Premature optimization, etc etc.

In theory, caching does make it faster.

In practice, benchmarks don't show it being faster, possibly because V8 already does that optimization internally. Either way, since it doesn't show up on benchmarks, I'd prioritize readability. Ditto re: looping backwards.